cheri_compressed_cap_common.h

```diff
@@ -108,10 +113,7 @@ typedef struct _cc_N(cap) _cc_N(cap_t);
 #define _cc_cap_t _cc_N(cap_t)
 
 static inline bool _cc_N(exactly_equal)(struct _cc_N(cap) const* a, struct _cc_N(cap) const* b) {
-    return a->cr_tag == b->cr_tag && a->_cr_cursor == b->_cr_cursor && a->cr_base == b->cr_base &&
-           a->_cr_top == b->_cr_top && a->cr_perms == b->cr_perms && a->cr_uperms == b->cr_uperms &&
-           a->cr_otype == b->cr_otype && a->cr_ebt == b->cr_ebt && a->cr_flags == b->cr_flags &&
-           a->cr_reserved == b->cr_reserved;
+    return a->cr_tag == b->cr_tag && a->_cr_cursor == b->_cr_cursor && a->cached_pesbt == b->cached_pesbt;
```

Comment:
Might want to assert that memcmp(a, b) == 0 if the pesbt+cursor match.

Reply:
I think that would break otherwise correct code. Comparing caps should not require decompressing. While pesbt and cursor are always valid, I think the core library should make no assumptions at the other fields, leaving consumers to call decompress when appropriate.

Reply:
memcmp is also not necessarily true if there's compiler-inserted padding, depending on how the structs were created